docs/snippets/gpu/preprocessing_single-plane.cpp

```diff
@@ -0,0 +1,55 @@
+#include <openvino/runtime/core.hpp>
+#define OV_GPU_USE_OPENCL_HPP
+#include <openvino/runtime/intel_gpu/ocl/ocl.hpp>
+#include <openvino/runtime/intel_gpu/properties.hpp>
+#include <openvino/core/preprocess/pre_post_process.hpp>
+
+ov::intel_gpu::ocl::ClImage2DTensor get_y_tensor();
+ov::intel_gpu::ocl::ClImage2DTensor get_uv_tensor();
+
+int main() {
+    ov::Core core;
+    auto model = core.read_model("model.xml");
+
+    //! [init_preproc]
+    using namespace ov::preprocess;
+    auto p = PrePostProcessor(model);
+    p.input().tensor().set_element_type(ov::element::u8)
+                      .set_ layout("NHWC")
+                      .set_memory_type(ov::intel_gpu::memory_type::surface);
+    p.input().model().set_layout("NCHW");
+    auto model_with_preproc = p.build();
+    //! [init_preproc]
+
+    auto compiled_model = core.compile_model(model, "GPU");
+    auto context = compiled_model.get_context().as<ov::intel_gpu::ocl::ClContext>();
+    auto input = model->get_parameters().at(0);
+    auto infer_request = compiled_model.create_infer_request();
+
+{
+    //! [single_batch]
+    cl::Image2D img_y_plane;
+    auto input_y = model_with_preproc->get_parameters().at(0); 
+    auto remote_y_tensor = remote_context.create_tensor(input_y->get_element_type(), in_shape, img_y_plane);
+    inf_req_remote.set_tensor(*input_y->output(0).get_tensor().get_names().begin(), remote_y_tensor);
+    inf_req_remote.infer() ;
+    //! [single_batch]
+}
+
+{
+    auto y_tensor_0 = get_y_tensor();
+    auto y_tensor_1 = get_y_tensor();
+    auto uv_tensor_0 = get_uv_tensor();
+    auto uv_tensor_1 = get_uv_tensor();
+    //! [batched_case]
+    cl::Image2D img_y_plane_0, img_y_plane_l;
+    auto input_y = model_with_preproc->get_parameters().at(0); 
+    auto remote_y_tensor_0 = remote_context.create_tensor(input_y->get_element_type(), in_shape, img_y_plane);
+    auto remote_y_tensor_1 = remote_context.create_tensor(input_y->get_element_type(), in_shape, img_y_plane);
+    std::vector<ov::Tensor> y_tensors = {remote_y_tensor_0, remote_y_tensor_1}; 
+    infer_request.set_tensors(*input_y->output(0).get_tensor().get_names().begin(), y_tensors);
+    infer_request.infer();
+    //! [batched_case]
+}
```

Comment:
why do we need new code snippets? is it extension of what we have?

Reply:
It is supposed to be an exact copy of the preprocessing.cpp snippet, with changes for single-plane input. If you see the option of putting all the code in one file, I'm all for it.
Even more so that there are some build issues... apparently the first line in this file breaks things... But in the original it does not.

Reply:
it's OK to have several files